src/message.c

```diff
@@ -1896,6 +1904,11 @@ msg_prt_line(char_u *s, int list)
 		c = lcs_trail;
 		attr = HL_ATTR(HLF_8);
 	    }
+	    else if (c == ' ' && lead != NULL && s < lead)
```

Comment:
On these lines, if a line is full of spaces, they are all _trailing spaces_, below (see other comment...)